x/valset-pref/validator_set_test.go

```diff
@@ -274,3 +205,132 @@ func (s *KeeperTestSuite) TestIsPreferenceValid() {
 		})
 	}
 }
+
+// NOTE: this is the case that used to error. Fixed by this PR
+func (s *KeeperTestSuite) TestUndelegateFromValSetErrorCase() {
+	s.SetupTest()
+
+	valAddrs := s.SetupMultipleValidators(2)
+	valPreferences := []types.ValidatorPreference{
+		{
+			ValOperAddress: valAddrs[0],
+			Weight:         sdk.NewDecWithPrec(5, 1), // 0.5
+		},
+		{
+			ValOperAddress: valAddrs[1],
+			Weight:         sdk.NewDecWithPrec(5, 1), // 0.5
+		},
+	}
+
+	delegator := sdk.AccAddress([]byte("addr1---------------"))
+	coinToStake := sdk.NewCoin(sdk.DefaultBondDenom, sdk.NewInt(10_000_000))   // delegate 10osmo using Valset now and 10 osmo using regular staking delegate
+	coinToUnStake := sdk.NewCoin(sdk.DefaultBondDenom, sdk.NewInt(20_000_000)) // undelegate 20osmo
+	expectedShares := []sdk.Dec{sdk.NewDec(15_000_000), sdk.NewDec(500_000)}
+
+	s.FundAcc(delegator, sdk.Coins{sdk.NewInt64Coin(sdk.DefaultBondDenom, 100_000_000)}) // 100 osmo
```

Comment:
I think generally for tests if we can easily know the amount the test case will use, we should be utilizing that amount. 

But this is just a personal opinion so you can disregard. I just know, it has more than once caught a bug where we were utilizing more funds than we were intending, and it was initially caught due to having insufficient funds despite funding with the amount that should have been used.

Reply:
implemented this feedback in tdd test: https://github.com/osmosis-labs/osmosis/pull/5967/files#diff-e8d94c443b26e1cda354d150b30ce5542344099e0bbbb8e84e507d9df791daf7R309